src/rules.ts

```diff
@@ -254,21 +234,61 @@ const isMatch: Matcher = (rule, options) => {
   return matches.length ? matches.every((match) => match === true) : false;
 };
 
-export const getMatchingRules = (
-  rules: Rule[],
-  files: RuleFile[],
-  event: Event,
-  patchContent: string[] = []
-): MatchingRule[] => {
-  const fileNames = files.map(({ filename }) => filename);
-
-  const matchingRules = rules.reduce((memo, rule) => {
-    if (isMatch(rule, { event, patch: patchContent, fileNames })) {
-      return [...memo, { ...rule, matched: true }];
-    } else {
-      return memo;
-    }
-  }, [] as MatchingRule[]);
+export class Rules extends Array<Rule> {
+  public constructor(...items: Rule[]) {
+    super(...items);
+  }
+  static loadFromLocation(location: string): Rules {
+    const matches = sync(location, {
+      onlyFiles: true,
+      cwd: env.GITHUB_WORKSPACE,
+      absolute: true,
+    });
 
-  return matchingRules;
-};
+    debug('files found:', matches);
+    const rules = matches.reduce((memo, filePath) => {
+      try {
+        const rule = loadJSONFile(filePath);
+
+        return isValidRawRule(rule) ? [...memo, { name: basename(filePath), ...sanitize(rule), path: filePath }] : memo;
+      } catch (e) {
+        console.error(`${filePath} can't be parsed, it will be ignored`);
+        return memo;
+      }
+    }, [] as Rule[]);
+
+    return new Rules(...rules);
+  }
+  getMatchingRules(files: RuleFile[], event: Event, patchContent?: string[]): MatchingRules {
+    return MatchingRules.load(this, files, event, patchContent);
+  }
+}
+
+class MatchingRules extends Array<MatchingRule> {
+  private constructor(...items: MatchingRule[]) {
+    super(...items);
+  }
+
+  @memo()
+  private groupByAction() {
+    return groupBy(this, (rule) => rule.action);
+  }
+  groupBy(action: keyof typeof RuleActions): MatchingRule[] {
+    const grouped = this.groupByAction()[action];
+    return grouped || [];
+  }
+
+  static load(rules: Rules, files: RuleFile[], event: Event, patchContent: string[] = []): MatchingRules {
```

Comment:
Similar question to that for `Rules.loadFromLocation`

Reply:
answered already in the previous one, but lemme know if is not the case.